testsuite/json-formatter.cc

```diff
@@ -0,0 +1,116 @@
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE snapper
+
+#include <boost/test/unit_test.hpp>
+
+#include <vector>
+#include <string>
+
+#include "../client/utils/JsonFormatter.h"
+
+using namespace std;
+
+BOOST_AUTO_TEST_CASE(test1_escape_values)
+{
+    snapper::cli::JsonFormatter::Data data = {
+	{ "key1", "value1" },
+	{ "key2", "value\"2" },
+	{ "key3", "value\\3" },
+	{ "key4", "\"value4\"" }
```

Comment:
A test case for a value containing a newline is missing. People might put newlines in snapshot descriptions. The current code just emits a literal newline, which is not allowed in a JSON string